pkg/services/nmap.go

```diff
@@ -281,6 +282,7 @@ func processNVDDataForPort(port nmap.Port, validCPE string) []tools.Vulnerabilit
 		slog.Any("error", err),
 	)
 
+	var cveIDs []string
```

Comment:
```suggestion
    cveIDs := make([]string, 0, len(nvdData.Vulnerabilities))
```

By initializing cveIDs with make([]string, 0, len(nvdData.Vulnerabilities)), you tell Go's runtime to pre-allocate an underlying array large enough to hold all potential CVE IDs from nvdData.Vulnerabilities. This can be useful to avoid unnecessary reallocs